packages/graphql-language-service-utils/src/__tests__/Range-test.js

```diff
@@ -0,0 +1,73 @@
+import {expect} from 'chai';
+import {describe, it} from 'mocha';
+import {Range, Position, offsetToPosition, locToRange} from '../Range';
+
+const text = `query test {
+  name
+}`;
+const absRange = {start: 15, end: 18}; //position of 'name' attribute in the test query
+const offsetRangeStart = new Position(1, 2);
+const offsetRangeEnd = new Position(1, 5);
+
+describe('class Position', () => {
+  it('constructs Position object', () => {
+    const pos = new Position(3, 5);
+    expect(pos).to.not.be.undefined;
+    expect(pos.character).to.equal(5);
+    expect(pos.line).to.equal(3);
+  });
+
+  it('compares Position object', () => {
+    const pos_1 = new Position(1, 2);
+    const pos_2 = new Position(2, 2);
+    const pos_3 = new Position(2, 3);
+    expect(pos_1.lessThanOrEqualTo(pos_2)).to.equal(true);
+    expect(pos_2.lessThanOrEqualTo(pos_3)).to.equal(true);
+    expect(pos_3.lessThanOrEqualTo(pos_1)).to.equal(false);
+  });
+});
+
+describe('class Range', () => {
+  const start = new Position(2, 3);
+  const end = new Position(2, 5);
+  const range = new Range(start, end);
```

Comment:
Typically these would be in a `before` function so that they get recreated before each `it` block. This is important for classes that may have internal mutable state, and even for classes that don't have it this is good hygiene.

```
let start;
let end;
let range;

before(() => {
  start = new Position(2, 3);
  end = new Position(2, 5);
  range = new Position(start, end);
});
```